com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/PyFunctionHashAdder/PyFunctionHashAdder_cpp.cgt

```diff
@@ -0,0 +1,168 @@
+/* Additional includes go here */
+
+#include <Python.h>
+#include <string>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <memory>
+
+#include "splpy.h"
+
+<%SPL::CodeGen::implementationPrologue($model);%>
+
+<%
+ my $attr =  $model->getInputPortAt(0)->getAttributeAt(0);
+ my $pystyle = 'unk';
+ if ($attr->getName() eq '__spl_po') {
+    $pystyle = 'python';
+ } elsif ($attr->getName() eq 'string') {
+    $pystyle = 'string';
+ } elsif ($attr->getName() eq 'jsonString') {
+    $pystyle = 'json';
+ }
+%>
+
+// Constructor
+MY_OPERATOR::MY_OPERATOR() : function_(NULL), pickleObjectFunction_(NULL)
+{
+  std::string tkDir = ProcessingElement::pe().getToolkitDirectory();
+  std::string streamsxDir = tkDir + "/opt/python/packages/streamsx/topology";
+  std::string splpySetup = streamsxDir + "/splpy_setup.py";
+  const char* spl_setup_py = splpySetup.c_str();
+
+  streamsx::topology::Splpy::loadCPython(spl_setup_py);
+
+  streamsx::topology::PyGILLock lock;
+
+    PyObject *_module_;
+    PyObject *_function_;
+
+    std::string appDirSetup = "import streamsx.topology.runtime\n";
+    appDirSetup += "streamsx.topology.runtime.setupOperator(\"";
+    appDirSetup += ProcessingElement::pe().getApplicationDirectory();
+    appDirSetup += "\")\n";
+
+    const char* spl_setup_appdir = appDirSetup.c_str();
+    if (PyRun_SimpleString(spl_setup_appdir) != 0) {
+         SPLAPPTRC(L_ERROR, "Python script splpy_setup.py failed!", "python");
+         PyErr_Print();
+         throw;
+    }
+
+<%
```

Comment:
Since the pr was first created the Python functional operators have been simplified to use common code.

Any chance of having this one align with the others?